randovania/games/prime_hunters/exporter/patch_data_factory.py

```diff
@@ -52,16 +61,53 @@ def force_field_index_for_requirement(game: GameDescription, requirement: Layout
     "Octolith8": 7,
 }
 
+_ITEM_TO_ITEM_TYPE = {
+    "Energy Tank": 4,
+    "Volt Driver": 5,
+    "Missile": 6,
+    "Battlehammer": 7,
+    "Imperialist": 8,
+    "Judicator": 9,
+    "Magmaul": 10,
+    "Shock Coil": 11,
+}
+
 
 class HuntersPatchDataFactory(PatchDataFactory[HuntersConfiguration, HuntersCosmeticPatches]):
     def game_enum(self) -> RandovaniaGame:
         return RandovaniaGame.METROID_PRIME_HUNTERS
 
-    def _calculate_starting_inventory(self, resources: ResourceCollection) -> dict[str, str]:
-        result = {
-            # TODO: Take starting weapons into account
-            "weapons_string": "00000101",
-        }
+    def _calculate_starting_inventory(self, resources: ResourceCollection) -> dict[str, str | int]:
+        result: dict[str, str | int] = {}
+        starting_items = {}
+
+        for resource, quantity in resources.as_resource_gain():
+            try:
+                starting_items[item_type_for_item_resource(resource)] = quantity
+            except KeyError:
+                print(f"Skipping {resource} for starting inventory: no item id")
+                continue
+
+        weapons = [
+            starting_items.get(_ITEM_TO_ITEM_TYPE["Shock Coil"], 0),
+            starting_items.get(_ITEM_TO_ITEM_TYPE["Magmaul"], 0),
+            starting_items.get(_ITEM_TO_ITEM_TYPE["Judicator"], 0),
+            starting_items.get(_ITEM_TO_ITEM_TYPE["Imperialist"], 0),
+            starting_items.get(_ITEM_TO_ITEM_TYPE["Battlehammer"], 0),
+            1,  # Missiles
+            starting_items.get(_ITEM_TO_ITEM_TYPE["Volt Driver"], 0),
+            1,  # Power Beam
+        ]
```

Comment:
then if you do as I suggested above, this would instead be:
```py
weapons = [
    starting_items["Shock Coil"],
    starting_items["Magmaul"],
    starting_items["Judicator"],
    starting_items["Imperialist"],
    starting_items["Battlehammer"],
    1, # Missiles
    starting_items["Volt Driver"],
    1, # Power Beam
]
```
much cleaner

Reply:
although actually it would be safer to do:
```py
def starts_with_weapon(weapon_name: str) -> bool:
    return starting_items[weapon_name] > 0

weapons = [
    starts_with_weapon("Shock Coil"),
    starts_with_weapon("Magmaul"),
    starts_with_weapon("Judicator"),
    starts_with_weapon("Imperialist"),
    starts_with_weapon("Battlehammer"),
    True, # Missiles
    starts_with_weapon("Volt Driver"),
    True, # Power Beam
]
```
that way if you somehow have multiple copies of a weapon in the starting inventory, things won't break